server/src/test/java/org/elasticsearch/action/admin/cluster/allocation/DesiredBalanceResponseTests.java

```diff
@@ -120,6 +159,39 @@ public void testToXContent() throws IOException {
         assertEquals(stats.get("computation_time_in_millis"), response.getStats().cumulativeComputationTime());
         assertEquals(stats.get("reconciliation_time_in_millis"), response.getStats().cumulativeReconciliationTime());
 
+        // cluster balance stats
+        Map<String, Object> clusterBalanceStats = (Map<String, Object>) json.get("cluster_balance_stats");
+        assertEquals(clusterBalanceStats.keySet(), response.getClusterBalanceStats().tiers().keySet());
+        for (var entry : response.getClusterBalanceStats().tiers().entrySet()) {
+            Map<String, Object> tierStats = (Map<String, Object>) clusterBalanceStats.get(entry.getKey());
+            assertEquals(Set.of("shard_count", "total_write_load", "total_shard_size"), tierStats.keySet());
+
+            Map<String, Object> shardCountStats = (Map<String, Object>) tierStats.get("shard_count");
+            assertEquals(Set.of("total", "average", "min", "max", "std_dev"), shardCountStats.keySet());
+            assertEquals(shardCountStats.get("total"), entry.getValue().shardCount().total());
```

Comment:
I was wondering if we can extract `entry.getValue` to a descriptive variable something like `var balanceStats = entry.getValue();`